### PR TITLE
Adding documentation

### PR DESCRIPTION
**Pull request by ryanmitchell**
Give this a once over and if you're happy I'll work on them all this week.

### CODE REVIEW

File: docs/customers.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
+## Customers
+
+This endpoint allows you to `list`, `create`, `retrieve`, `update` and `delete` customers on your Tasty Igniter site.
```
Comment (sampoyigi) [inline docs/customers.md:3]: TastyIgniter not Tasty Igniter :)

File: docs/customers.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+### The customer object
+
+#### Attributes
+
+| Key                  | Type      | Description                                                  |
+| -------------------- | --------- | ------------------------------------------------------------ |
+| `first_name`           | `string`  | **Required**. The customer's first name (between 2 and 32 characters in length)      |
```
Comment (sampoyigi) [inline docs/customers.md:11]: Use 48 here as it will be the next max limit for names in upcoming versions

File: docs/customers.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+| `last_name`           | `string`  | **Required**. The customer's last name (between 2 and 32 characters in length)       |
+| `full_name`           | `string`  | A concatenation of first_name and last_name       |
+| `email`           | `string`  | **Required**. The customer's email address       |
+| `telephone`           | `string`  | The customer's telephone number         |
+| `date_added`           | `datetime`  | The date and time the customer was added to your site       |
+| `newsletter`           | `boolean`  | Whether the customer opts into newsletter marketing         |
+| `customer_group_id`           | `integer`  | The group the customer belongs to, if any.         |
+| `status`           | `boolean`  | Has the value `true` if the customer is enabled or the value `false` if the customer is disabled.         |
+| `addresses`           | `array`  | The customer's addresses, if any        |
+
```
Comment (sampoyigi) [inline docs/customers.md:21]: Orders and reservations relation are also available on the customer object

File: docs/customers.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+
+#### Payload example
+
+```json
+{
+  "first_name": "Joe",
+  "last_name": Blogs,
```
Comment (sampoyigi) [inline docs/customers.md:92]: typo

File: docs/customers.md [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+
+Updates a customer.
+
+```
+PATCH /api/customers/:customer_id
+```
+
+#### Parameters
+
+| Key                  | Type      | Description                                                  |
```
Comment (sampoyigi) [inline docs/customers.md:260]: Lets use one or two parameters as sample for the update endpoint, to keep things simple.

**Comment by ryanmitchell** [comment]
All changes in place

**Comment by sampoyigi** [comment]
Looking good 👍

**Comment by ryanmitchell** [comment]
Great. Leave the rest with me and I’ll work on them this week.

**Comment by sampoyigi** [comment]
All done, please review for errors and typos before I merge.

**Comment by ryanmitchell** [comment]
A couple of minor changes and additions.

On orders - should there not be a way of adding order_menus?

**Comment by sampoyigi** [comment]
Thank you. Agreed, adding order_menus should be available, do you want to help with that?

**Comment by ryanmitchell** [comment]
Yeah I can work on that - what do you think the best approach is?

**Comment by sampoyigi** [comment]
You can follow the approach used within `Admin\Traits\ManageOrderItems::addOrderMenus()`.

**Comment by ryanmitchell** [comment]
Thats helpful thanks.

So it could be as simple as calling $model->addOrderMenus() on the store() and update() methods of the orders resource when a request key of menu_items is present? Or do you think it needs to have its own separate logic borrowing from addOrderMenus() ?

**Comment by sampoyigi** [comment]
Yep a separate logic since the `addOrderMenus` method only accepts a CartContent instance, or simply change the `addOrderMenus` logic to accept an array (pass `Cart::content()->all()` instead) to avoid duplicate codes.

**Pull request merged**